pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -48,6 +48,9 @@ in
     preBuild = optionalString stdenv.isDarwin ''
       sed -i -e "s|tr1/type_traits|type_traits|g" \
       -e "s|std::tr1|std|" src/util.h
+    '' + ''
+      make -j${NIX_BUILD_CORES} -l${NIX_BUILD_CORES} -C $out mksnapshot
```

Comment:
I don't think there's a way, except to copy and edit [that line from stdenv](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/setup.sh#L702) or similar.